File: src/java/org/javaturk/dp/ch05/adapter/electricity/domain/tr/TurkishPowerProvider.java
package org.javaturk.dp.ch05.adapter.electricity.domain.tr;

public class TurkishPowerProvider implements TurkishPowerSource {
	
	public TurkishPowerProvider(){
		System.out.println("TurkishPowerProvider is up and running.");
	}

	@Override
	public void providePowerAt220V() {
		System.out.println("I provide electricity at 220V. Be careful, there may be some casual interruptions!");
	}

	@Override
	public void turnOn() {
		System.out.println("TurkishPowerProvider started to provide electricity.");
	}

	@Override
	public void turnOff() {
		System.out.println("TurkishPowerProvider stopped to provide electricity.");
	}
}

File: src/typescript/com/gokselkucuksahin/dp/structural-patterns/adapter/electricity/domain/tr/turkish-power-provider.ts
import {TurkishPowerSource} from "./turkish-power-source";

export class TurkishPowerProvider implements TurkishPowerSource {
  constructor() {
    console.log("TurkishPowerProvider is up and running.");
  }

  providePowerAt220V(): void {
    console.log("I provide electricity at 220V. Be careful, there may be some casual interruptions!");
  }

  turnOff(): void {
    console.log("TurkishPowerProvider stopped to provide electricity.");
  }

  turnOn(): void {
    console.log("TurkishPowerProvider started to provide electricity.");
  }

}